lib/index.js

```diff
@@ -108,357 +131,286 @@ exports.isBoom = function (err, statusCode) {
 };
 
 
-exports.boomify = function (err, options) {
+exports.boomify = function (err, options = {}) {
 
-    Hoek.assert(err instanceof Error, 'Cannot wrap non-Error object');
+    const { override, data, decorate, statusCode, message } = options;
 
-    options = options || {};
-
-    if (options.data !== undefined) {
-        err.data = options.data;
+    if (!err?.isBoom === true) {
+        return new exports.Boom(message, { statusCode, cause: err, data, decorate });
```

Comment:
The return type here does not seem to match the definition:
```ts
Terr extends Boom ? Terr : Boom<Data>
```
Indeed, the type signature implies that if there is no `data` in option, and err is a Boom, the returned type would be `Boom<Original>`, but here, in the implementation, it seems that the original `data` & `decoration` are not preserved.